### PR TITLE
Convert the parameter classes to the `.from_et()` pattern

### PR DESCRIPTION
**Pull request by andlaus**
IMO this cleans up the parameter construction code significantly. On the flipside it requires to make the `Parameter` class instantiable...

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/parameters/createanyparameter.py [hunk 1]
```diff
@@ -27,205 +23,33 @@
...
-        dop_snref = None
-        if (dop_snref_elem := et_element.find("DOP-SNREF")) is not None:
-            dop_snref = odxrequire(dop_snref_elem.get("SHORT-NAME"))
-
-        if dop_ref is None and dop_snref is None:
-            raise ValueError(
-                f"A parameter of type {parameter_type} must reference a DOP! {dop_ref}, {dop_snref}"
-            )
-
     if parameter_type == "VALUE":
```
Comment (kayoub5) [inline odxtools/parameters/createanyparameter.py:29]: instead of multiple ifs, maybe use a lookup dict?

Comment (kayoub5) [reply odxtools/parameters/createanyparameter.py:29]: other than this, LGTM

Comment (andlaus) [reply odxtools/parameters/createanyparameter.py:29]: that would make things harder to read and I doubt that the performance would be measurably better...

**Comment by andlaus** [comment]
ok, let's merge this. Thanks for the review @kayoub5!

**Pull request merged**